### PR TITLE
update tidb tutorial to use st.experimental_connection

### LINKED ISSUE #633: Update TiDB tutorial to use st.experimental_connection
### Problem

We recently launched [st.experimental_connection](https://blog.streamlit.io/introducing-st-experimental_connection/) and support for community connections.

Given that we had past interest in connecting Streamlit to TiDB, it might be useful to update the [current tutorial](https://docs.streamlit.io/knowledge-base/tutorials/databases/tidb) to use this approach.

Per discussion in https://github.com/pingcap/docs/issues/13396 the recommended approach is to use the built in SQLConnection with the [TiDB SQLAlchemy driver](https://github.com/pingcap/sqlalchemy-tidb) assuming we verify it works easily enough.

### Additional context (if relevant)

Connections can be built by extending [ExperimentalBaseConnection](https://docs.streamlit.io/library/api-reference/connections/st.connections.experimentalbaseconnection). We've built a few examples:

- https://github.com/streamlit/files-connection
- https://github.com/streamlit/gsheets-connection
- https://github.com/streamlit/release-demos/blob/master/1.22/st-experimental-connection/duckdb_connection/connection.py

If someone is interested in taking this on and setting up the repo, would be happy to discuss, do code review, or set up a quick call to discuss.

(Same goes for other relevant data stores / data sources!)

### PR DESCRIPTION
**Pull request by zhangyangyu**
## 📚 Context

close #633 

## 🧠 Description of Changes

**Revised:**

_Insert screenshot of your updated UI/code here_

**Current:**

_Insert screenshot of existing UI/code here_

## 💥 Impact

Size:

- [x] Small 
- [ ] Not small 

## 🌐 References

- [ ] [Notion](...)

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

### CODE REVIEW

**Comment by sfc-gh-jcarroll** [comment]
Will leave the formal review to @snehankekre but the updates look good to me generally.

File: content/kb/tutorials/databases/tidb.md [hunk 1]
```diff
@@ -101,43 +106,47 @@ Copy the code below to your Streamlit app and run it. Make sure to adapt `query`
...
+
+[connections.tidb]
+dialect = "mysql"
+driver = "pymysql"
+host = "<TiDB_cluster_host>"
+port = 4000
+database = "pets"
+username = "<TiDB_cluster_user>"
+password = "<TiDB_cluster_password>"
+create_engine_kwargs = { connect_args = { ssl = { ca = "<path_to_CA_store>" }}}
```
Comment (sfc-gh-jcarroll) [inline content/kb/tutorials/databases/tidb.md:151]: Oh this is interesting, I haven't tried creating the nested dict like this before in toml

@zhangyangyu did you confirm this works with the example app?

Comment (zhangyangyu) [reply content/kb/tutorials/databases/tidb.md:151]: Yes. I tried. Nested dict in toml could only be on one line, easy to be wrong :-).

**Review by snehankekre — APPROVED** [review]
These changes look good to me too! Thanks for updating the tutorial, @zhangyangyu 😄

**Pull request merged**